matchzoo/preprocessors/units/tokenize.py

```diff
@@ -15,3 +17,104 @@ def transform(self, input_: str) -> list:
         :return tokens: tokenized tokens as a list.
         """
         return nltk.word_tokenize(input_)
+
+
+class ChineseTokenize(Unit):
+    """Process unit for text containing Chinese tokens."""
+
+    def transform(self, input_: str) -> str:
+        """
+        Process input data from raw terms to processed text.
+
+        :param input_: raw textual input.
+
+        :return output: text with a blank between adjacent Chinese tokens.
+        """
+        output = []
+        for char in input_:
+            cp = ord(char)
+            if is_chinese_char(cp):
+                output.append(" ")
+                output.append(char)
+                output.append(" ")
+            else:
+                output.append(char)
+        return "".join(output)
+
+
+class BasicTokenize(Unit):
+    """Process unit for text tokenization."""
+
+    def transform(self, input_: str) -> list:
+        """
+        Process input data from raw terms to list of tokens.
+
+        :param input_: raw textual input.
+
+        :return tokens: tokenized tokens as a list.
+        """
+        orig_tokens = whitespace_tokenize(input_)
+        split_tokens = []
+        for token in orig_tokens:
+            split_tokens.extend(run_split_on_punc(token))
+
+        output_tokens = whitespace_tokenize(" ".join(split_tokens))
+        return output_tokens
```

Comment:
What's the difference between this one and the NLTK one?

Reply:
In `nltk` package, `TreebankWordTokenizer` is used and words like `can't` are tokenized as `ca` and `n't`. But in bert-based model, this process should be executed by the `WordPieceTokenize` to avoid improper tokenization results.